pkg/store/proxy.go

```diff
@@ -381,14 +416,39 @@ func (s *streamSeriesSet) Err() error {
 	return errors.Wrap(s.err, s.name)
 }
 
-// matchStore returns true if the given store may hold data for the given label matchers.
+// matchStore returns true if the given store may hold data for the given label
+// matchers.
 func storeMatches(s Client, mint, maxt int64, matchers ...storepb.LabelMatcher) (bool, error) {
 	storeMinTime, storeMaxTime := s.TimeRange()
 	if mint > storeMaxTime || maxt < storeMinTime {
 		return false, nil
 	}
+	return labelSetsMatch(s.LabelSets(), matchers)
+}
+
+// labelSetsMatch returns false if each label-set negative matches against the
```

Comment:
I would change this to (find it more clear for me), but up to you (: 
```suggestion
// labelSetsMatch returns false if all label-set does not match the
```